java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -279,7 +279,7 @@ private predicate elementSpec(
 bindingset[namespace, type, subtypes]
 private RefType interpretType(string namespace, string type, boolean subtypes) {
   exists(RefType t |
-    t.hasQualifiedName(namespace, type) and
+    [t, t.getSourceDeclaration()].hasQualifiedName(namespace, type) and
```

Comment:
I think this is the wrong place to put `.getSourceDeclaration()`. Instead we should change the 3 instances of `ref.(Call).getCallee() = e` to `ref.(Call).getCallee().getSourceDeclaration() = e` below in `sourceElementRef`, `sinkElementRef`, and `summaryElementRef`. With this change we'll track much fewer tuples in the intermediate predicates. It also allows specifying signatures when type variables are mentioned, as the current solution doesn't work for that case.

Reply:
Done